Initial Docs (#70)
streamshub/flink-sql: docs/deploying.adoc

For details on what Flink SQL can do, please refer to the https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/table/overview/[upstream documentation].

=== Using secrets

You can use Kubernetes secrets, with the Flink SQL Runner, to provide security credentials to Flink job for connecting to the source or the target systems.
Secrets can be directly templated in the SQL statements with the following pattern:

[source,yaml]
----
{{secret:<NAMESPACE>/<SECRET NAME>/<DATA KEY>}}
----

=== Special Characters

Note that semicolon `+;+` is a special character used as a statement delimiter. If it's part of your SQL statements, make sure it is escaped by `+\\+`. 
For example, it might be used for `+properties.sasl.jaas.config+` value when using Kafka connector. 
In this case, it would look something like this:

[source,yaml]
----
'properties.sasl.jaas.config' = 'org.apache.flink.kafka.shaded.org.apache.kafka.common.security.plain.PlainLoginModule required username=\"test-user\" password=\"{{secret:flink/test-user/user.password}}\"\\;'
----

== Deploying a Flink SQL Query

Your SQL query can be submitted via the `+spec.job.args+` field of the `+FlinkDeployment+` CR.

> It should be formed of a single string within an array literal (`+[ ]+`), https://yaml-multiline.info/[multi-line yaml strings] (via the use of `+|+`,`+>+` characters) are not currently supported. 

Reviewer: ```suggestion
It should be formed of a single string within an array literal (`+[ ]+`).
https://yaml-multiline.info/[Multi-line yaml strings] (using `+|+`,`+>+` characters) are not currently supported. 
```

> However, within the single string newline, tab and other whitespace characters will be ignored, so queries can be properly formatted.

Reviewer: ```suggestion
However, newlines, tabs, and other whitespace characters within a single string are ignored, so queries can still be well-formatted.
```

> See the example below for an illustration of this.

Reviewer: ```suggestion
See the example below for an illustration of the formatting.
```
